<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/convert_sequences">May 16, 2021 - Extract nucleotide and amino acid sequences from genbank and ali</a></h1>
            <div class="post-content">
            <p>In bioinformatics, it is common to extract information as there are multiple formats. First let us look at easy one. One of the requests was to extract the sequences from .ali into a fasta formatted sequence file. .ali files were taken from <a href="https://biologydirect.biomedcentral.com/articles/10.1186/1745-6150-4-29#additional-information">here</a>. I donâ€™t understand the format though it is named with .ali extension. In general, .ali extensions are for <code class="language-plaintext highlighter-rouge">PIR</code> format and the sequences were not in <code class="language-plaintext highlighter-rouge">PIR</code> format. Let us jump to the code. Here is the code for extracting sequences in .fasta format, from 13062_2009_161_MOESM2_ESM.ali.</p>

<pre><code class="language-code">$ tr -s $'\t' &lt; 13062_2009_161_MOESM2_ESM.ali | awk -v OFS="\n" '{print "&gt;"$1,$2}' &gt; 13062_2009_161_MOESM2_ESM.fa
</code></pre>
<p>Now let us get to genbank to fasta. Since genbank can have both nucleotide sequences and amino acid sequences (CDS), I wrote a code to extract both the sequences (Nucleotide and aminoacid) and is deposited in github. You can download it from <a href="https://github.com/svsuresh/python_scripts/blob/13d1d66e046f3ae2e320e3a330376fda2e56aa12/gbtofasta/gbtofasta.py">here</a>. Input file is hard coded and example files are included in the same directory as script. There is another use case where user wants to extract CDS sequence (amino acid), but wants to filter the sequence by a gene name, <a href="https://github.com/svsuresh/python_scripts/blob/13d1d66e046f3ae2e320e3a330376fda2e56aa12/gbtofasta/gbk2seq.py">here</a> is the code to filter and extract sequence from genebank file. This script takes the user input for genbank and output file. If output file is name is not furnished, input extension is removed and <code class="language-plaintext highlighter-rouge">.fasta</code> extension will be appended. For eg <code class="language-plaintext highlighter-rouge">test.fasta</code> will be generated from <code class="language-plaintext highlighter-rouge">test.gbk</code>. Following is the usage.</p>

<pre><code class="language-code">./gbk2seq.py input.gbk
./gbk2seq.py input.gbk output.fa
</code></pre>
<p>In fist case, output file name would be <code class="language-plaintext highlighter-rouge">input.fasta</code> and in later case, output would be <code class="language-plaintext highlighter-rouge">output.fa</code></p>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/identify_contiguous_blocks" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/convert_tab_to_fasta" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

