<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/convert-vcf-chrom-column-from-hgxx">Sep 7, 2015 - Convert VCF #CHROM column from hgxx format to b3x format</a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on">
In general, genome coordinates in VCF and reference fasta files will be in either hgxx (for eg hg19, hg38) or b3x (b32, b35) 
format.<br />
<br />
<div style="text-align: justify;">
hgxx format will have "chr" prefix to each chromosome number and Mitochondrial (chrM) chromosome will be at the start. b3x format will have no prefix for chromosome numbers and Mitochondrial chromosome will be at the last. If one has used hg19 reference (fasta from UCSC) in variant calling ( and prior steps in NGS analysis) and would like to annotate using clinvar/dbSNP, user has to append string "chr" to the first column "#CHROM" in VCF. To append string "chr" to chromosome numbers in #CHROM column, use the following command:</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
command:</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
sed 's/^\([1-9,X,Y,MT]\)/chr&amp;/g' clinvar_GRCh38_12jul2015.vcf &gt; clinvar_12jul2015_b38.vcf</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
One can download clinvar vcf file from ftp://ftp.ncbi.nlm.nih.gov/snp/organisms/human_9606_b142_GRCh38/VCF/&nbsp; for GRCh 38.</div>
<br />
Before:<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="http://4.bp.blogspot.com/--0ftJWoq-68/Ve09LaSoUdI/AAAAAAAAAYQ/BZ78yrRRlA0/s1600/before.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="316" src="http://4.bp.blogspot.com/--0ftJWoq-68/Ve09LaSoUdI/AAAAAAAAAYQ/BZ78yrRRlA0/s640/before.png" width="640" /></a></div>
<br />
After:<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="http://2.bp.blogspot.com/-DL1kbHnwdKw/Ve09S8piSEI/AAAAAAAAAYY/B9LCj8iD2fc/s1600/after.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="336" src="http://2.bp.blogspot.com/-DL1kbHnwdKw/Ve09S8piSEI/AAAAAAAAAYY/B9LCj8iD2fc/s640/after.png" width="640" /></a></div>
<br /></div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/download-and-use-microarray-files-from" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/rnaseq-data-analysis-and-tuxedo-workflow" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

