<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/download-and-use-microarray-files-from">Sep 10, 2015 - Download and use microarray files from GEO in R</a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on">
GEO is repository service offered by NCBI to make genomics and transcriptomics data available to public. There are several ways to download the data from GEO. From R, here is the simple script. To download data from GEO within R, one needs package, by name "<a href="http://bioconductor.org/packages/release/bioc/html/GEOquery.html" target="_blank">GEOquery</a>". This note <br />
<br />
Following is the code to download data (CEL files)<br /><br />
#######################################<br />
##GEO Query script to fetch cel files#############<br />#######################################<br />
library(GEOquery)<br />gse8771 = getGEOSuppFiles("GSE8771")<br />untar("GSE8771/GSE8771_RAW.tar", exdir="data")<br />cels=list.files("data/", pattern = "CEL")<br />sapply(paste("data", cels, sep="/"), gunzip)<br />##################################################<br /><br />
getGEOsuppFiles downloads all the cel files as a single tar file and stores it in a new folder with name "GSExxxx" in the current working directory. In the example code, R creates a new folder titled "GSE8771", in current working directory. Files are downloaded as single tar file with gzipped sample files inside, to save bandwidth. User can untar the files either in the current directory or directory of choice. Example code stores untarred archive will with gzipped sample data in a folder titled "data" in current working directory. These files can be gunzipped in a batch, using sapply function. Once gunzipped, user can read the files using functions in "simpleaffy" package <br /></div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/transcriptomics-microarray-data-analysis" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/convert-vcf-chrom-column-from-hgxx" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

