<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/extract-genes-samples-from-cluster-of">Jul 16, 2018 - Gene clustering and extract clusters, genes and samples of interest</a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on">
<div style="text-align: justify;">
Many a times, very simple tasks seem to be difficult for eg. extracting members of a cluster. For a biologist, clustering of genes or pathways or samples very common. Then the question comes how to extract the genes of a particular cluster. Let us say you have a dendrogam with 7 clusters and you would like to extract genes of cluster 5.<br />
Let us do a small simulation and then extract genes from a cluster of interest. Remember you can also do the same for any cluster or for any other variable (for eg sample, pathway, network module).</div>
===================================<br />
&gt; data &lt;- replicate(20, rnorm(100, mean =10, sd=10))<br />
&gt; rownames(data) &lt;- paste("Gene", c(1:nrow(data)))<br />
&gt; colnames(data) &lt;- paste("Sample", c(1:ncol(data)))<br />
&gt; library(stats) <br />
&gt; d &lt;- dist(data, method = "euclidean")<br />
&gt; hc &lt;- hclust(d, method = "ward.D2")<br />
&gt; plot(hc)<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="https://3.bp.blogspot.com/-z587cWPpijs/W0zeuu3eohI/AAAAAAAABKY/b49djfiElxMwK0bCObf3OfXFPD3_RAHYQCLcBGAs/s1600/Rplot01.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" data-original-height="755" data-original-width="969" height="497" src="https://3.bp.blogspot.com/-z587cWPpijs/W0zeuu3eohI/AAAAAAAABKY/b49djfiElxMwK0bCObf3OfXFPD3_RAHYQCLcBGAs/s640/Rplot01.png" width="640" /></a></div>
<div style="text-align: justify;">
Now how do we extract the genes of interest? To do that, first we need to define the number of clusters in such a way that genes of interest fall in an independent cluster. How do we do that? We keep partitioning dendrogram till we identify the cluster with genes of interest. We can partition the data&nbsp; by defining a limit/boundary on height of dendrogram. For eg. in above diagram, if we cut the tree at 130 , we are left with three clusters (count always from the top, highlighted in red color below)</div>
<div class="separator" style="clear: both; text-align: center;">
<a href="https://1.bp.blogspot.com/-Zh2MOznFOgk/W-cIeVtUJeI/AAAAAAAAAyc/TRQP-6pk1eMRNoO74u_MUYSbawwBlhfUwCLcBGAs/s1600/Rplot02.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" data-original-height="794" data-original-width="624" height="640" src="https://1.bp.blogspot.com/-Zh2MOznFOgk/W-cIeVtUJeI/AAAAAAAAAyc/TRQP-6pk1eMRNoO74u_MUYSbawwBlhfUwCLcBGAs/s640/Rplot02.png" width="502" /></a></div>
Now let us say,&nbsp; we cut the tree at 110, we will have 5 clusters.<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="https://2.bp.blogspot.com/-3fg1KEYj_SY/W-cPBLhLN8I/AAAAAAAAAy8/IYB_dd0GjrIQ9j-4vXzrB6So1Ke_13aDACLcBGAs/s1600/Rplot.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" data-original-height="794" data-original-width="624" height="640" src="https://2.bp.blogspot.com/-3fg1KEYj_SY/W-cPBLhLN8I/AAAAAAAAAy8/IYB_dd0GjrIQ9j-4vXzrB6So1Ke_13aDACLcBGAs/s640/Rplot.png" width="502" /></a></div>
<div class="separator" style="clear: both; text-align: justify;">
Now we know how to partition the dendrogram. After partitioning the dendrogram, we can extract the genes from the cluster of interest. For eg. let us say we want to extract genes from 5th cluster after partitioning the data with a height of 110.&nbsp;</div>
<div class="separator" style="clear: both; text-align: justify;">
Let us extract all the genes for which cluster is assigned 5 and their expression values. </div>
============================================== <br />
# Cut the tree at 110<br />
h=110<br />
k=cutree(hc,h=h)<br />
# Extract the genes assigned to cluster 5<br />
names(k[k==5])<br />
&nbsp;# Now extract the expression values&nbsp; for the genes in cluster 5<br />
data[names(k[k==5]),]<br />
===============================================<br />
&nbsp;(Note: Dendrogram produced above may not be same each time we run the code as data is randomly simulated. Set seed to get consistent dendrograms)</div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/replce-wobble-base-with-purine-r-and" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/add-sample-number-to-venn-diagram" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

