<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/fetch-chromosome-sizes-from-ucsc">Jul 18, 2014 - Fetch chromosome sizes from UCSC in linux </a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on">
In ubuntu, chromsizes ca be fetched multiple ways:<br />
<br />
Method 1: <br />
<br />
In GATK pipeline, user has to provide chromosome sizes. For this,&nbsp; "fetchChromSizes" utility from UCSC helps.&nbsp; User can download it from here: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/ <br />
<br />
To fetch chromosome sizes for build hg38, user should run following command:<br />
<br />
$ sh fetchChromSizes &lt;UCSC genome build&gt; &lt;output_file&gt;<br />
<br />
Example: $ sh fetchChromSizes hg38 &gt; hg38.sizes (for hg38).<br />
<br />
Same information can be found by visiting the URL:<br />
<br />
http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&amp;chromInfoPage=.<br />
<br />
For any other build, change db=hg38 with appropriate genome build (For eg. dm3 will give Drosophila genome size and url would be: http://genome.ucsc.edu/cgi-bin/hgTracks?<b>db=dm3</b>&amp;chromInfoPage=).<br />
<br />
2) Method 2:<br />
<br />
UCSC, by default provides chromsizes from direct URL<br />
<br />
For eg. for hg19, link would be http://genome.ucsc.edu/goldenpath/helpc and for hg 39, link would be&nbsp; http://genome.ucsc.edu/goldenpath/help/hg38.chrom.sizes.<br />
<br />
toget it direct in linux, run the following command:<br />
<br />
wget -np -nd -r http://genome.ucsc.edu/goldenpath/help/hg38.chrom.sizes<br />
<br />
-np, -nd tells the command that do not create any additional directories, download the file to the current director and -r instructs the command to fetch the file recursively.<br />
<br />
Interesting number of entries in each build, kept on increasing. Look at the following picture: (compares between hg18.chrom.sizes,hg19.chrom.sizes and hg38.chrom.sizes).<br />
<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="http://4.bp.blogspot.com/-ZpMi6pt52MQ/VXFFTzaq0lI/AAAAAAAAAOY/AZQuyuP1BdQ/s1600/chromsizes.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="150" src="http://4.bp.blogspot.com/-ZpMi6pt52MQ/VXFFTzaq0lI/AAAAAAAAAOY/AZQuyuP1BdQ/s640/chromsizes.png" width="640" /></a></div>
<br />
<br />
</div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/gene-expressoin-and-treemaps" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/sort-and-merge-fasta-files-by" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

