<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/find-common-genes-between-two-gene-lists">Aug 6, 2017 - Find common genes between two gene lists</a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on">
There are times, we come across to<br />
<ol style="text-align: left;">
<li>Find common genes between two pathways/lists</li>
<li>Find genes that are unique to list 1 excluding common genes between two lists</li>
<li>Find genes that are unique to list 2, excluding common genes between two lists </li>
</ol>
<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="https://2.bp.blogspot.com/-gS_nG5rUiLg/WYYkwcev36I/AAAAAAAAAiw/XPOvrL1MplkXd0PNQVbtpJh11p3piRNzwCLcBGAs/s1600/Untitled%2B2.jpg" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" data-original-height="338" data-original-width="554" height="244" src="https://2.bp.blogspot.com/-gS_nG5rUiLg/WYYkwcev36I/AAAAAAAAAiw/XPOvrL1MplkXd0PNQVbtpJh11p3piRNzwCLcBGAs/s400/Untitled%2B2.jpg" width="400" /></a></div>
<br />
Following note is an R code examples to do above tasks. For this you would need to R (grab latest version) and&nbsp; dplyr package (again, grab latest package).<br />
<br />
Before we dive into the code execution, let me briefly explain join function we will be using below. Join function does what it says. It joins.&nbsp; When we say join, we need to tell program how to join two gene lists (L1 and L2). Both the gene lists (L1 and L2) should have same column names for gene lists and other columns. For eg. If list 1 has gene names under column named "Genes", list2 column with gene names should have title "Genes".<br />
<br />
When joining, there can be three types of joining: Genes&nbsp; common between two lists (called inner_join),Genes unique to list 1 only (exclude common genes and list2 genes) and Genes unique to list 2 only (exclude common genes and list2 genes). Common genes are those genes which are common between list1 and list 2. Now let us get to work:<br />
<br />
Let us assume that we have two gene lists: L1 and L2 with following genes and they also have expression trends.<br />
<br />
------------------------------------------------------------------------------<br />
L1<br />
------------------------------------------------------------------------------<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; gene&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Expression<br />
<ol style="text-align: left;">
<li><i>RARA</i>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; up</li>
<li><i>RNF40</i>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down</li>
<li><i>RUVBL2</i>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down</li>
<li><i>SETD1A</i>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down</li>
<li><i>ACTR8</i>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down</li>
</ol>
------------------------------------------------------------------------------<br />
L2<br />
------------------------------------------------------------------------------<br />
<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; gene&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Expression<br />
<ol style="text-align: left;">
<li>ACTL6A&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down</li>
<li>ACTL6B&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; up</li>
<li>ACTR5&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; up</li>
<li>ACTR8&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down</li>
</ol>
&nbsp;-------------------------------------------------------------------------------<br />
<br />
-------------------------------------------------------------------------------<br />
<b>Common genes between L1 and L2:</b><br />
------------------------------------------------------------------------------<br />
Let us find out the common genes between L1 and L2:<br />
<b> </b><br />
<b>_____________________________________________</b><br />
<br />
<b>Code <br />______________________________________________ </b><br />
<b>$ library(dplyr)</b><br />
<b>$ inner_join(L1,L2)<br />&nbsp;</b><br />
<b>Output:</b><br />
<b>&nbsp;Joining, by = c("gene", "Expression")<br />&nbsp;&nbsp;&nbsp; gene&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Expression <br />1 ACTR8&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down </b><br />
==================================================<br />
<br />
--------------------------------------------------------------------------------<br />
<b>Genes unique to L1 (exclude common genes and genes in L2)</b><br />
<b>-------------------------------------------------------------------------------- </b><br />
Let us find out the unique genes to L1:<br />
<b> </b><b>_____________________________________________</b><br />
<br />
<b>Code <br />______________________________________________ </b><br />
<b>$ library(dplyr)</b><br />
<b>$ anti_join(L1,L2) #Note L1 is on left side<br />&nbsp;</b><br />
<b>Output:</b><br />
<b>&nbsp;Joining, by = c("gene", "Expression")<br />&nbsp;&nbsp;&nbsp; gene&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Expression </b><br />
<b>1 SETD1A&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down<br />2 RUVBL2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down<br />3 RNF40&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down<br />4 RARA&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; up</b><br />
<b><br /></b>
<br />
==================================================<br />
<br />
----------------------------------------------------------------------------------<br />
<b>Genes unique to L2 (exclude common genes and genes in L1)</b><br />
-----------------------------------------------------------------------------------<br />
Let us find out the unique genes to L2:<br />
<b> </b><b>_____________________________________________</b><br />
<br />
<b>Code <br />______________________________________________ </b><br />
<b>$ library(dplyr)</b><br />
<b>$ anti_join(L2,L1) #Note L1 is on right side<br />&nbsp;</b><br />
<b>Output:</b><br />
<b>&nbsp;Joining, by = c("gene", "Expression")<br />&nbsp;&nbsp;&nbsp; gene&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Expression </b><br />
<b>1&nbsp; ACTR5&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; up<br />2 ACTL6B&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; up<br />3 ACTL6A&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; down</b><br />
<b><br /></b></div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/split-and-melt-columns-in-r" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/remove-both-rows-and-columns-with-na" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

