<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/heat-map-with-rnaseq-featurecounts">Nov 29, 2017 - Heat map with RNAseq Featurecounts output</a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on">
<div style="text-align: justify;">
There are several tutorials on generating heatmaps using microarray, Exon array and tools that generate differential data. In this note, let us explore simplest way to create heatmap with RNAseq data. Post DESeq2 analysis (RNASeq HISAT2-Featurecounts-DESeq2 work flow outlined <a href="https://digibio.blogspot.in/2017/11/rna-seq-analysis-hisat2-featurecounts.html" target="_blank">here</a>), you might have arrived at a list of genes with statistical significance with expression (fold change cutoff). For generating heatmap, do following:<br />
<br />
<ol>
<li>Extract genes (features of interest). From DESeq2 one would get Fold change, p-value, adjusted p-value etc. Now extract only genes</li>
<li>Subset read count (normalized and available within DESeq2) for these genes only. </li>
</ol>
</div>
<ol style="text-align: justify;"></ol>
<div style="text-align: justify;">
You can download the file from here: This data is DESeq2 normalized. There is another component needed i.e sample demography i.e grouping information such as age, condition, gender etc. Let us deal with very simple one i.e samples with one grouping information: Disease (Turmor and Normal). This meta information is available for these samples. Sample Names are in columns and genes are in rows. Column names of expression data ( with read counts) should match that from row names of demography data as the sample names should match between both the data (expression data and demography data). In R parlance, column names of expression data should match the row names of sample demography (meta data). In this example, you can observe the same.<br />
<br /></div>
<div style="text-align: justify;">
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; condition<br />
hcc1395_normal_rep1&nbsp;&nbsp;&nbsp; normal<br />
hcc1395_normal_rep2&nbsp;&nbsp;&nbsp; normal<br />
hcc1395_normal_rep3&nbsp;&nbsp;&nbsp; normal<br />
hcc1395_tumor_rep1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; tumor<br />
hcc1395_tumor_rep2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; tumor<br />
hcc1395_tumor_rep3&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; tumor</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
What we are going to do is to plot the expression values and group the samples by condition.</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
code (I am not showing how the data is loaded into R. It comes from either analysis with in R or from an external file):</div>
<div style="text-align: justify;">
===================================================</div>
<div style="text-align: justify;">
$ library(pheatmap)</div>
<div style="text-align: justify;">
$ pheatmap(as.matrix(log.rc.fc.sig.lfcshrink_res_ddds_ordered), scale = "row", clustering_distance_rows = "correlation", clustering_method = "complete",color = c("green", "red"),annotation_col = samples, labels_col&nbsp; = str_replace(rownames(samples),"hcc.[1-9]+_",""), main="Significant genes", fontsize_col=24, fontsize_row = 4)</div>
<div style="text-align: justify;">
================================================================</div>
<div style="text-align: justify;">
Explanation of the command:</div>
<ol style="text-align: left;">
<li>as.matrix - convert the data to matrix</li>
<li>scale - scale the data by row</li>
<li>clustering_distance_rows - now the samples must be clustered by correlation. Distance estimate can be changed for eg. by Euclidean distance</li>
<li>Clustering_method - there are&nbsp; 2-3 methods. How clustering can be done: for eg complete, average, nearest etc. Here, it is complete</li>
<li>Color - color gradient for expression values. Here I have used only two. You can use as many as you want. But the graph becomes too clumsy. Use intuition in color selection.</li>
<li>annotation_col - this is very important. Now we want to column (samples) data to be clustered as per the condition. For this data we provide is sample meta data data frame.</li>
<li>labels_col - column labels. We can supply any name that we wish to. But they should match with number of columns.&nbsp; In the above code, we are chopping of sample names so that they fit well in graph</li>
<li>main - title of the graph</li>
<li>fontsize_col - column font size (For sample names, I put large number, as they are small in number)</li>
<li>fontsize_row - row font size (For gene names, I put small number as they are many in number)</li>
</ol>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
</div>
<div style="text-align: justify;">
Heatmap would like this:</div>
<div style="text-align: justify;">
<br /></div>
<div class="separator" style="clear: both; text-align: center;">
<a href="about:invalid#zClosurez" imageanchor="1" style="clear: right; float: right; margin-bottom: 1em; margin-left: 1em;"></a><a href="https://4.bp.blogspot.com/-Kc8mbeVXfqU/Wh5gh5_75aI/AAAAAAAAA4w/sbdJBIOEkYMDJX2XNpuNSQH5crmi5Ue6wCLcBGAs/s1600/Rplot.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" data-original-height="812" data-original-width="1200" height="432" src="https://4.bp.blogspot.com/-Kc8mbeVXfqU/Wh5gh5_75aI/AAAAAAAAA4w/sbdJBIOEkYMDJX2XNpuNSQH5crmi5Ue6wCLcBGAs/s640/Rplot.png" width="640" /></a></div>
<div style="text-align: justify;">
<br /></div>
<div class="separator" style="clear: both; text-align: center;">
<br /></div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
<br /></div>
</div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/nucleotide-base-frequency-per-read-and" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/rna-seq-analysis-hisat2-featurecounts" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

