<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/natural-sort-chromosome-number-in-r">Jun 5, 2015 - natural sort chromosome number in R - method 1</a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on">
Some times, trivial issues bother. Here is a small tip to sort a chromosomes (values in a column) in human readable chronological order. This sorting is also called alpha, numeric sort. Following note, gives a general example to sort a data frame in natural order based on any user mentioned column.<br />
<br />
Requirements:<br />
package "naturalsort" in R<br />
<br />
Commands:<br />
<br />
1. data.frame[naturalorder(vector in dataframe), ] (by default, the values are sorted ascending order)<br />
Note: This would sort the data frame in natural sort as per the values in mentioned vector in data frame, in ascending order.<br />
<br />
2. data.frame[naturalorder((vector in dataframe), decreasing=TRUE), ]<br />
(by default, the values are sorted ascending order) <br />
<br />
Note: This would sort the data frame in natural sort as per the values in metioned vector in data frame, in descending order.<br />
<br />
Example:<br />
<br />
1. chr.hg19[naturalorder(chr.hg19$V1),]<br />
This would sort chrosomes from chromosome 1 to chromsome XY. <br />
<br />
2.&nbsp; chr.hg19[naturalorder((chr.hg19$V1),decreasing=TRUE),]<br />
This would sort chrosomes from chromosome XY to chromsome 1. <br />
Note: chr.hg19 is data frame created from http://genome.ucsc.edu/goldenpath/help/hg19.chrom.sizes and taking only chromosomes. <br />
<br />
Case study:<br />
1) Import chromosome sizes from UCSC<br />
2) Sort the chromosomes in human readable order. <br />
<br />
Method: <br />
<br />
1)&nbsp; Load library "naturalsort" in R.<br />
<br />
2) Download the data and store it as object in R. <br />
command: chr.hg19=read.delim(url("http://genome.ucsc.edu/goldenpath/help/hg19.chrom.sizes"),&nbsp; <br />
<br />
3) Print chr.hg19<br />
command: dim(chr.hg19)<br />
<br />
This would display tha there are 93 rows and with 2 columns. First colum contains chromosomes, other artefacts. Column two contains lengths of the chromosomes in columns in bp.<br />
<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="http://4.bp.blogspot.com/-xj0yFijXyFE/VXE08O2KPeI/AAAAAAAAAN8/XaNxCCVCqhg/s1600/orig_chr.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="640" src="https://4.bp.blogspot.com/-xj0yFijXyFE/VXE08O2KPeI/AAAAAAAAAN8/XaNxCCVCqhg/s640/orig_chr.png" width="377" />&nbsp;</a></div>
<div class="separator" style="clear: both; text-align: center;">
<br /></div>
<div class="separator" style="clear: both; text-align: left;">
3) Since we deal with chromosomes only, we need to extract only chromosomes and their coordinates. We know that chromosomes are 24 (22 autosomes + XY). </div>
<div class="separator" style="clear: both; text-align: left;">
<br /></div>
<div class="separator" style="clear: both; text-align: left;">
Command: chr29=chr.hg19[1:24,] </div>
<div class="separator" style="clear: both; text-align: left;">
This would save first 24 entries/chromosomes as a data.frame. Please note that chromosomes are sorted numerically (column V1).</div>
<div class="separator" style="clear: both; text-align: center;">
<br /></div>
<div class="separator" style="clear: both; text-align: center;">
&nbsp;<a href="http://2.bp.blogspot.com/-bo07SJ-QVpQ/VXE07SfqgWI/AAAAAAAAANw/5wCIVnJFerY/s1600/chrom1.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="640" src="https://2.bp.blogspot.com/-bo07SJ-QVpQ/VXE07SfqgWI/AAAAAAAAANw/5wCIVnJFerY/s640/chrom1.png" width="280" />&nbsp;</a></div>
<div class="separator" style="clear: both; text-align: left;">
4) Sort chromosomes in human/alpha numeric order, in descending order i.e from chr1 to chrx and y.&nbsp;</div>
command:&nbsp; chr.hg19.asc=chr.hg19[naturalorder(chr.hg19$V1),]<br />
<div class="separator" style="clear: both; text-align: left;">
</div>
<div class="separator" style="clear: both; text-align: center;">
<br /></div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
<a href="http://3.bp.blogspot.com/-6n0aiYyWVX0/VXE07flY55I/AAAAAAAAAN4/5SPThIpMoQc/s1600/chr1_inc.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="640" src="https://3.bp.blogspot.com/-6n0aiYyWVX0/VXE07flY55I/AAAAAAAAAN4/5SPThIpMoQc/s640/chr1_inc.png" width="499" /></a></div>
<br />
5) Now, for some reason, if you need to sort in reverse order (ie. descending fashion), following is the command:<br />
<br />
&nbsp;command: chr.hg19[naturalorder((chr.hg19$V1),decreasing=TRUE),]<br />
&nbsp; <br />
<div class="separator" style="clear: both; text-align: center;">
<a href="http://1.bp.blogspot.com/-aHrSL003uBQ/VXE07e4G2sI/AAAAAAAAAN0/bE5aWNHtBAk/s1600/chr_dec.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="554" src="https://1.bp.blogspot.com/-aHrSL003uBQ/VXE07e4G2sI/AAAAAAAAAN0/bE5aWNHtBAk/s640/chr_dec.png" width="640" /></a></div>
<br />
All commands:<br />
<br />
library(naturalsort) ## loads the library naturalsort<br />
chr.hg19=read.delim(url("http://genome.ucsc.edu/goldenpath/help/hg19.chrom.sizes"), header=FALSE) ## downloads the chromomosome sizes and stores it as object "chr.hg19"<br />
chr.hg19=chr.hg19[1:24,] # extracts first 24 lines (chromosomes)<br />
chr.hg19.asc=chr.hg19[naturalorder(chr.hg19$V1),] ## sorts the chromosomes in natural order in ascending fashion<br />
chr.hg19.desc=chr.hg19[naturalorder((chr.hg19$V1), decreasing=TRUE),] ##&nbsp; sorts the chromosomes in natural order in descending fashion.<br />
<br />
<br />
<br />
<br /></div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/exporting-ideogram-information-from" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/plot-and-high-light-genes-of-interest" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

