<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/plotting-variants-per-chromosome-using">May 21, 2016 - Plotting variants per chromosome using ggplot2</a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on">
<a href="https://4.bp.blogspot.com/-GCTYVtCedSc/V0CT9gVE7CI/AAAAAAAAAeg/J-i-ASzDeNcsjTmYasd4PGAPE1BqOdWwQCLcB/s1600/all_variants_r.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"></a>There are several methods&nbsp; of representing data esp biological data to end users. Some of them are complex and some, simple. I am not a fan of complex data representations as the underlying data needs comprehension of complex data or incorrect representation of simple data. However, once in a while, such complex figures may be necessary to understand the data. Following one and this image represents the variants per chromosome (SNV, MNV and DIVs). I would like user to go through this <a href="http://digibio.blogspot.com/2015/07/parsing-vcf-2-clinvar-using-dplyr-and.html" target="_blank">topic</a> in the blog to parse VCF into tab file and subsequent data manipulations. I have used clinvar VCF for data and used bcftools to convert VCF to tab file. For graphics, I have used ggplot2.<br />
<br />
Object of this tutorial to is represent variants present in <a href="https://drive.google.com/file/d/0B0MpwluEDxNuSnF3M0FVUDZtWTQ/view?usp=sharing" target="_blank">clinvar VCF</a>. Please note that the latest clinvar VCF can be downloaded from <a href="ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz" target="_blank">here</a> for GRCh 38. Variants are categorized under SNV, MNV (Structural variants in general) and DIV (indels) categories and are represented by VC (variant category) VCF.&nbsp; Please follow the instructions on this <a href="http://digibio.blogspot.com/2015/07/parsing-vcf-2-clinvar-using-dplyr-and.html" target="_blank">blog note</a>, to load vcf data into R.<br />
<br />
I present two images here: 1) Variants presents in all the chromosomes<br />
2) Variants present per chromosome<br />
<br />
First let us list all the variants in clinvar (data frame) using:<br />
<br />
1) Summarize all the variants across all chromosomes:<br />
<br />
$ clinvar&nbsp; %&gt;% group_by(VC) %&gt;% summarise(Variants=n())<br />
<br />
(Note: use dplyr package for this).<br />
<br />
Result would be:<br />
<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="https://3.bp.blogspot.com/-GCTYVtCedSc/V0CT9gVE7CI/AAAAAAAAAeg/0t16dFVNj_8dHv77ZoMBZguHUcjMmd_kQCKgB/s1600/all_variants_r.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="206" src="https://3.bp.blogspot.com/-GCTYVtCedSc/V0CT9gVE7CI/AAAAAAAAAeg/0t16dFVNj_8dHv77ZoMBZguHUcjMmd_kQCKgB/s640/all_variants_r.png" width="640" /></a></div>
<br />
Please note that MNVs are much small in number in compared to both DIV (indels) and SNVs.<br />
<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="https://4.bp.blogspot.com/-08v6QbZ9RpQ/V0CS0V_qZoI/AAAAAAAAAeU/VcKN6eltUY0f1YGLdi8Y8k82ioYnBbOmwCLcB/s1600/all_variants.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="412" src="https://4.bp.blogspot.com/-08v6QbZ9RpQ/V0CS0V_qZoI/AAAAAAAAAeU/VcKN6eltUY0f1YGLdi8Y8k82ioYnBbOmwCLcB/s640/all_variants.png" width="640" /></a></div>
To visualize the same in R using ggplot2, run commands given below: <br />
<br />
$ qplot(CHROM, data=clinvar, fill=VC) + <br />&nbsp; theme(<br />&nbsp;&nbsp;&nbsp; panel.background=element_rect(colour = "black", fill=NA), <br />&nbsp;&nbsp;&nbsp; strip.background=element_blank(), <br />&nbsp;&nbsp;&nbsp; panel.grid.major = element_blank()) + <br />&nbsp; facet_wrap(~VC,scales = "free")<br /><br />
Now, let us draw all the chromosomes individually and corresponding variants as shown below:<br />
1) Let us draw all variants (SNVs, MNVs and DIVs). Please note that you won't be seeing MNVs as they are very very less in number compared to DIVs and SNVs per chromosome.<br />
<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="https://3.bp.blogspot.com/-rLYL_hFl-8w/V0CV6UwYI6I/AAAAAAAAAes/TQ5vyihZiP0Ccj1cInyQjR-JJeK7BBI6gCLcB/s1600/variants_per_chromosome.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="412" src="https://3.bp.blogspot.com/-rLYL_hFl-8w/V0CV6UwYI6I/AAAAAAAAAes/TQ5vyihZiP0Ccj1cInyQjR-JJeK7BBI6gCLcB/s640/variants_per_chromosome.png" width="640" /></a></div>
<br />
Code for getting above image is:<br />
<br />
$ ggplot(clinvar, aes(CHROM, fill=VC))+<br />&nbsp; geom_bar(position = "dodge")+ <br />&nbsp; facet_wrap(~CHROM,ncol=5,nrow = 5, scales = "free")+<br />&nbsp; theme(<br />&nbsp;&nbsp;&nbsp; strip.background=element_blank(), <br />&nbsp;&nbsp;&nbsp; panel.background = element_rect(colour = "black", fill=NA), <br />&nbsp;&nbsp;&nbsp; axis.text.x =element_blank(), <br />&nbsp;&nbsp;&nbsp; axis.ticks.x =element_blank(),<br />&nbsp;&nbsp;&nbsp; panel.grid.major = element_blank())+ <br />&nbsp; labs(title = "Variants in Clinvar",&nbsp; x="Chromosome", y="Number of variants per chromosome")<br />
<br />
Listing all the variants per chromosome is:<br />
$ clinvar&nbsp; %&gt;% group_by(CHROM,VC) %&gt;% summarise(Variants=n())<br />
<br />
Example output is:<br />
<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="https://4.bp.blogspot.com/-u2lph6-U-04/V0CXFXLDeRI/AAAAAAAAAe8/VPqW7XXXVMcR4gxboBvvdBOeqdJn8GZKACLcB/s1600/variantsperchromosome.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="422" src="https://4.bp.blogspot.com/-u2lph6-U-04/V0CXFXLDeRI/AAAAAAAAAe8/VPqW7XXXVMcR4gxboBvvdBOeqdJn8GZKACLcB/s640/variantsperchromosome.png" width="640" /></a></div>
<br />
<br />
<br /></div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/generate-random-dna-elements-using" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/vcf-tricks" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

