<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/replce-wobble-base-with-purine-r-and">Jul 20, 2018 - Replace wobble base with purine (R) and pyrimidine (Y) symbol in python</a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on">
<div style="text-align: justify;">
Most biologists know that a codon represents an amino acid and is composed of 3 contiguous bases. Of the three bases, third base shows limited variability. Because of this an amino acid can be encoded by more than one codon. One of the requests I got was to convert every third base to corresponding purine (R) and pyrimidine (Y) symbol. As per IUPAC nucleotide code, bases A and G are purines and bases C and T are pyrimidines. Let us do this in python without any external library. I am using python 3 for this task. First let us take a simple fasta with arbitrary number of sequences&nbsp; and convert that into a sequence file with R and Y at every third base.</div>
<br />
input:<br />
========================================<br />
$ cat sequence.fa <br />
&gt; gene1 <br />
ACATATTGGAGGCCGAAACAATGAGGCGTGATCAACTCAGTATATCAC<br />
&gt; gene2 <br />
CTAACCTCTCCCAGTGTGGAACCTCTATCTCATGAGAAAGCTGGGATGAG<br />
&gt; gene3 <br />
ATTTCCTCCTGCTGCCCGGGAGGTAACACCCTGGACCCCTGGAGTCTGCA<br />
========================================<br />
<br />
output:<br />
========================================<br />
$ cat output.fa <br />
&gt;gene1<br />
ACRTAYTGRAGRCCRAARCARTGRGGYGTRATYAAYTCRGTRTAYCAY<br />
&gt;gene2<br />
CTRACYTCYCCYAGYGTRGARCCYCTRTCYCAYGARAARGCYGGRATRAG<br />
&gt;gene3<br />
ATYTCYTCYTGYTGYCCRGGRGGYAAYACYCTRGAYCCYTGRAGYCTRCA<br />
============================================<br />
<br />
code (note that input and output files are hard coded. change it as per requirement):<br />
======================================<br />
&gt; d = {}<br />
&gt; with open("sequence.fa", 'r') as f:<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; for i in f:<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; d[i.strip("\n&gt; ")] = next(f).rstrip()<br />
<br />
&gt; ntdict = {"A": "R", "T": "Y", "G": "R", "C": "Y"}<br />
<br />
&gt; with open("output.fa", "w") as f:<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; for key, value in d.items():<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; temp_key = "".join(ntdict.get(w.upper(), w) if not i % 3 else w for i, w in enumerate(list(d[key]), 1))<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; print("{0}{1}\n{2}".format("&gt;", key, temp_key.upper()), file=f)<br />
============================================<br />
Ps: Take care of indents.<br />
<br /></div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/lookaround-in-r-for-non-word-and-word" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/extract-genes-samples-from-cluster-of" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

