<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    09 August 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/split_fasta_by_header_regex">Split fasta file into multiple files based on partial headers</a></li>
        
          <li><a href="/add_filenames_to_all_the_columns_in_bulk">Append file name at the end of each column for all CSV files</a></li>
        
          <li><a href="/add_duplicate_count_to_headers">Add count to duplicated IDs in fasta file</a></li>
        
          <li><a href="/remove_bases">Remove bases in between the sequences</a></li>
        
          <li><a href="/retain_unique_values">Retain only unique values in the data</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/volcano-plot-with-ggplot2-and-basic">May 12, 2018 - Volcano plot with ggplot2 and basic plotting</a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on" zoompage-fontsize="16">
<div style="text-align: justify;" zoompage-fontsize="16">





</div>
<div class="container-fluid main-container" style="text-align: justify;" zoompage-fontsize="16">











<div class="fluid-row" id="header" style="text-align: justify;" zoompage-fontsize="16">




</div>
<div class="section level3" id="volcano-plot" style="text-align: justify;" zoompage-fontsize="16">
<h3 zoompage-fontsize="19">
Volcano plot</h3>
Volcano plot is not new. In the era of microarrays, they were used in
 conjunction with MA plots. Volcano plot is a plot between p-values 
(Adjusted p-values, q-values, -log10P and other transformed p-values) on
 Y-axis and fold change (mostly log2 transformed fold change values) on 
X-axis. Then one adds all kinds of decorations to plot like cut-off 
lines so and so forth.&nbsp; In this note, we will use basic plotting and ggplot2 to plot volcano plots. You can download the input file from<a href="https://docs.google.com/document/d/12nZ2qukpVmurKdU5BQIkyLJNBOCmctkILvWuwVyBLA8/edit?usp=sharing" zoompage-fontsize="16"> here</a> and the data has 3 columns: Gene names, Fold change (log2) and Adjusted p values (FDR). File is comma separated.<br />

</div>

<div class="section level3" id="basic-plotting" style="text-align: justify;" zoompage-fontsize="16">
<h3 zoompage-fontsize="19">
Basic plotting</h3>Plot will have fold change (log2) values on x-axis and adjusted p values (FDR) (-log10 scale) will be on Y-axis. Significant genes are upregulated or downregulated by 1.5 times (log 2 scale 0.6). Non-siginficant genes are hightlighted in grey color and significant genes are highlighted in red color. Genes are labeled with grey color. FDR cut off value is 0.05.&nbsp; Let us jump into the code. Red lines represent cut offs. P value cut offs (0.01 and 0.05) are log transformed. <br /></div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">================================================================</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16"># Load the data</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">df=read.csv("ggplot_test.txt", header = T, stringsAsFactors = F, strip.white = T)</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16"># Remove the empty lines</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">df=na.omit(df)</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16"># Plot the x and y axis with fold change and -log10 pvalues</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">with(df, <br /></div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">plot (FoldChange_log2, -log10(padj),</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">col = "darkgrey",</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">pch = 19,</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">cex=2,</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">xlab=expression("Fold Change, Log"[2]*""),</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">ylab=expression("Adjusted P value, Log"[10]*""),</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">abline(v=c(-0.6,0.6),h=c(-log10(0.01),-log10(0.05)), col="red", lty=3,lwd=3)))</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16"># Highlight the genes of interest</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">with(subset(df,abs(FoldChange_log2) &gt;= 0.6 &amp; padj &lt; 0.05 ),</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">points(</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">x = FoldChange_log2,</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">y = -log10(padj),</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">col = "red",</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">pch = 16,</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">cex=2))</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16"># Label the genes with steel blue color</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">with(subset(df,abs(FoldChange_log2) &gt;= 2 &amp; padj &lt; 0.05 ),</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">text(</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">x = FoldChange_log2,</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">y = -log10(padj),</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">Gene.names,</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">cex = 1,</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">pos=2,</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">col = "steelblue"</div><div class="section level3" style="text-align: justify;" zoompage-fontsize="16">))<br />

</div><div class="separator" style="clear: both; text-align: center;" zoompage-fontsize="16"><a href="https://1.bp.blogspot.com/-0UNr4BrU8cY/XtQf3TRoofI/AAAAAAAABXk/z9Un8j9rzjI-peBZ0Pl7At9yagY-6xNegCK4BGAsYHg/baseplot.png" style="margin-left: 1em; margin-right: 1em;" zoompage-fontsize="16"><img border="0" data-original-height="983" data-original-width="1920" height="328" src="https://1.bp.blogspot.com/-0UNr4BrU8cY/XtQf3TRoofI/AAAAAAAABXk/z9Un8j9rzjI-peBZ0Pl7At9yagY-6xNegCK4BGAsYHg/w640-h328/baseplot.png" width="640" /></a></div>
<div class="section level3" id="with-ggplot2" style="text-align: justify;" zoompage-fontsize="16">
<h3 zoompage-fontsize="19">
With ggplot2</h3>
<div class="section level4" id="load-libraries" zoompage-fontsize="16">
<h4 zoompage-fontsize="16"><span style="font-weight: normal;" zoompage-fontsize="16">===========================================================</span><br /></h4>
<pre class="r" zoompage-fontsize="12"><code class="hljs" zoompage-fontsize="12">suppressPackageStartupMessages(<span class="hljs-keyword" zoompage-fontsize="12">library</span>(dplyr))
suppressPackageStartupMessages(<span class="hljs-keyword" zoompage-fontsize="12">library</span>(ggplot2))
suppressPackageStartupMessages(<span class="hljs-keyword" zoompage-fontsize="12">library</span>(ggrepel))</code></pre>
</div>ggplot(df, aes(x = FoldChange_log2, y = -log10(padj),color = ifelse(abs(FoldChange_log2)&gt;0.6,"red","grey"))) +<br /><div class="section level4" id="now-let-us-plot-the-data-in-ggplot2" zoompage-fontsize="16">&nbsp;&nbsp;&nbsp; geom_point() +<br />&nbsp;&nbsp;&nbsp; xlab(expression("Fold Change, Log"[2]*"")) +<br />&nbsp;&nbsp;&nbsp; ylab(expression("Adjusted P value, Log"[10]*"")) +<br />&nbsp;&nbsp;&nbsp; geom_vline(<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; xintercept = c(-0.6,0.6),<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; col = "red",<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; linetype = "dotted",<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; size = 1) +<br />&nbsp;&nbsp;&nbsp; geom_hline(<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; yintercept = c(-log10(0.01),-log10(0.05)),<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; col = "red",<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; linetype = "dotted",<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; size = 1)+<br />&nbsp;&nbsp;&nbsp; theme_bw() +<br />&nbsp;&nbsp;&nbsp; theme(legend.position = "none")+<br />&nbsp;&nbsp;&nbsp; scale_colour_manual(values = c("grey", "red")) +<br />&nbsp;&nbsp;&nbsp; geom_text_repel(data=subset(df,abs(FoldChange_log2) &gt;= 2 &amp; padj &lt; 0.05),<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; aes(FoldChange_log2, -log10(padj), label = Gene.names),size = 3, color="steelblue")</div><div class="section level4" zoompage-fontsize="16">=======================================================<br /></div><div class="section level4" zoompage-fontsize="16"><br /></div>
</div>
</div><div class="separator" style="clear: both; text-align: center;" zoompage-fontsize="16"><a href="https://1.bp.blogspot.com/-JEYVnyXzms4/XtQfsDN2uHI/AAAAAAAABXY/-Mk2yHC0lFosDalybNgMa1_m7nw649dGwCK4BGAsYHg/ggplot2.png" style="margin-left: 1em; margin-right: 1em;" zoompage-fontsize="16"><img border="0" data-original-height="983" data-original-width="1920" height="328" src="https://1.bp.blogspot.com/-JEYVnyXzms4/XtQfsDN2uHI/AAAAAAAABXY/-Mk2yHC0lFosDalybNgMa1_m7nw649dGwCK4BGAsYHg/w640-h328/ggplot2.png" width="640" /></a></div>
</div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/concatenate-two-columns-in-r" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/manipulate-fasta-headers-and-gtf" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

