---
layout: post
title: Parsing Clinvar VCF using dplyr and vcflib
date: '2015-07-13T19:02:00.002+05:30'
author: suresh
tags:
- clinvar
- dplyr
- bcf tools
modified_time: '2015-07-13T19:13:55.756+05:30'
thumbnail: http://1.bp.blogspot.com/-5NbXgJfjP0o/VaOshltL_iI/AAAAAAAAARg/xIL8fUGu6mk/s72-c/varcat_nolog.png
blogger_id: tag:blogger.com,1999:blog-3175511529662794199.post-7144671738336795028
blogger_orig_url: https://digibio.blogspot.com/2015/07/parsing-vcf-2-clinvar-using-dplyr-and.html
---

<div dir="ltr" style="text-align: left;" trbidi="on">
Parsing and annotation of statistically significant variants is very important in NGS analysis. Statistically variants are stored in standard format, well known as VCF. A variant of this format is called gVCF (genomic VCF) which stores base information at non-variant locations as well.<br />
<br />
There are several public annotations available for variants identified in NGS studies. One of the best annotation is available as clinvar.vcf including curated polymorphic and clinically significant variants &nbsp; In this note, we use clinvar vcf and dplyr to extract information and we will use vcflib package to convert vcf to tab separated file.<br />
<br />
In linux as well as windows, there are several good tools such as bcftools,&nbsp; vcflib to extract information from VCF. There are several tools available in R to manipulate VCF files using gRanges. However, we take an easy route in this note with following steps:<br />
<br />
1) Download clinvar vcf file from NCBI or the&nbsp; example clinvar vcf ( ~ 6MB file and can be downloaded from <a href="https://drive.google.com/file/d/0B0MpwluEDxNuSnF3M0FVUDZtWTQ/view?usp=sharing" target="_blank">here</a>). (MD5sum: debad87790852ae518956ef0f3f3d453)<br />
2) Convert clinvar vcf to tab separated file using vcf2tsv function available in package vcflib<br />
3) Import tab separated file into R and analyze the data using dplyr. <br />
<br />
Following commands are necessary for the steps mentioned above:<br />
1) Download <a href="ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz" target="_blank">clinvar vcf</a> (for hg38) from NCBI or from example<a href="http://debad87790852ae518956ef0f3f3d453/" target="_blank"> clinvar.vcf</a> link. ( 6MB file, MD5sum: debad87790852ae518956ef0f3f3d453)<br />
2) Index gzipped VCF file. This would create a new file with ".tbi" extension which is index for the the vcf file.<br />
command: tabix -p vcf &lt;input.vcf&gt;<br />
Example: tabix -p vcf clinvar_GRCh38_12jul2015.vcf.gz <br />
Please note that VCF file and index should be in the same directory for further processing and tabix program is part of vcflib<br />
3) Convert vcf to tab separate values (tsv) file<br />
command: vcf2tsv input.vcf.gz &gt; output.tab<br />
Example: vcf2tsv clinvar_GRCh38_12jul2015.vcf.gz &gt; clinvar_GRCh38_12jul2015.vcf.tab<br />
<br />
Cross checking VCF output:<br />
1) Please note that ALT column in VCF file can have more than one allele. If there are more than one allele in ALT column, vcf2tsv creates row for each individual allele. Hence rows (records) in tab separated file must be more than vcf records.<br />
2) In later steps, we are going to count single nt variations (SNVs), multi nt variations (MNV) and indels (DIVs). These must match with those from vcf statistics generated from bcf tools.<br />
To generate vcf statistics, use following command:<br />
command: bcftools stats input.vcf.gz&nbsp; &gt;&nbsp; output.stats<br />
ExampleL bcftools stats clinvar_GRCh38_12jul2015.vcf.gz&nbsp; &gt; clinvar_GRCh38_12jul2015.vcf.stats<br />
<br />
This would create a text file with total number of records, number of SNVs, MNVs, DIVs,&nbsp; transition/ transvertion ratio and other information related to VCF. we are going to match the numbers between output for dplyr and bcfstats, to cross check the output.<br />
<br />
With clinvar.vcf in hand, we can ask several questions:<br />
1) what is the most represented chromosome i.e chromosome with highest number of clinival variants?<br />
2) What is the most represented diseases in clinivar.vcf?<br />
3) What is the number of&nbsp; SNVs MNVs and DIVs in clinvar? <br />
4) Which gene has highest number of variants ?<br />
5) List of all variants sorted by genes (in decreasing order)..like wise.<br />
<br />
We will try to answer few questions here with code. Rest questions (the ones you have) can be answered by adapting the code given below.<br />
<br />
1) Before that let us load the tab separated values in R.<br />
<br />
code: clinvar=read.delim2("input.tab", header=TRUE, sep="\t")<br />
Example: clinvar=read.delim2("clinvar_GRCh38_12jul2015.vcf.tab", header=TRUE, sep="\t")<br />
<br />
2) Let us check if all the entries are imported comparing lines (records) in input file and clinvar object in R.<br />
<br />
code: dim(clinvar)&nbsp; # prints the length and variables in the clinvar.&nbsp; Please note down the length of the object.<br />
<br />
In linux, open a terminal and type the following command:<br />
code: wc -l &lt;input.tab&gt; # This would print total number of records in the input file.<br />
Example: wc -l clinvar_GRCh38_12jul2015.vcf.tab<br />
<br />
Compare the numbers. Since original file has headers, the two numbers differ by one header line ( for eg. 134207 in linux Vs 134206 in R).<br />
<br />
3) One can also compare the headers of the file and column names of R object. Code is given below<br />
<br />
Command in R:&nbsp; colnames(clinvar)<br />
Header in linux: head -1 &lt;input.tab&gt;<br />
Example in linux: head -1 clinvar_GRCh38_12jul2015.vcf.tab<br />
<br />
Compare the two outputs, should be identical.<br />
<br />
4) Load the following libraries, which we will be using in analysis and graphing<br />
<br />
Command:<br />
lib.pac= c("dplyr", "ggplot2", "ggvis")<br />
lapply (lib.pac, require, character.only=TRUE)<br />
<br />
<br />
This code would load all the 3 packages at the same time. If they do not exist, please install them. Arguments (commands) for dplyr will follow a logic outlined below:<br />
<br />
data.frame &lt;then&gt; execute the commands &lt;then&gt; execute the commands &lt;then&gt; execute the commands. "then" is represented by "%&gt;%"<br />
<br />
5) List variants per chromosome (all SNVs, DIVs, MNVs together per chromosome)<br />
<br />
Command: total.var.per.chr=clinvar&nbsp; %&gt;% group_by(CHROM) %&gt;% summarise(n=n())<br />
<br />
6) View the first five lines<br />
Command: head(total.var.per.chr)<br />
<br />
These lines would give total number of variants per chromosome.&nbsp; Please note that chromosomes are not ordered alpha numerically/naturally/human sorted.<br />
<br />
7) List all variants per each category of variants<br />
<br />
Command: var.cat=clinvar&nbsp; %&gt;% group_by(VC) %&gt;% summarise(Variants=n())<br />
<br />
View all variants per variant category <br />
Command: var.cat&nbsp; <br />
<br />
7)&nbsp; List variant categories per chromosome i.e list SNVs, DIVs and MNVs separately for each chromosome <br />
<br />
Command: var.per.chr=clinvar&nbsp; %&gt;% group_by(CHROM,VC) %&gt;% summarise(n=n())<br />
<br />
8) Print the first few lines of var.per.chr<br />
<br />
Command:&nbsp; head (var.per.chr)<br />
<br />
Lines would list SNVs, MNVs, DIVs&nbsp; individuall for each chromosome.<br />
<br />
9) List only SNVs. Please note that this command can be adapted for DIVs and MNVs as well.<br />
<br />
Command: clinvar.snv=clinvar %&gt;% filter(VC=="SNV") %&gt;% group_by(CHROM,VC) %&gt;% summarise(n=n())<br />
<br />
Please note that we are filtering column "VC" by "SNV". We can combine multiple terms here.<br />
<br />
10) View the first few lines<br />
<br />
Command:&nbsp; head(clinvar.snv)<br />
<br />
11) Visualize the variants per chromosome. Please note that MNVs are very few compared to other categories. So one cannot visualize them without transforming the number of variants.&nbsp; Below, are two graphs : one with original numbers, the other with log (10) transformed values. We will be using ggvis for visualization. We can also use ggplot2. Code for ggplot2 is given at the end of the note.<br />
<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="http://1.bp.blogspot.com/-5NbXgJfjP0o/VaOshltL_iI/AAAAAAAAARg/xIL8fUGu6mk/s1600/varcat_nolog.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="140" src="http://1.bp.blogspot.com/-5NbXgJfjP0o/VaOshltL_iI/AAAAAAAAARg/xIL8fUGu6mk/s200/varcat_nolog.png" width="200" /></a><a href="http://1.bp.blogspot.com/-Oi7ISfakhCU/VaOshgn7WsI/AAAAAAAAARc/uc62xuTPQHw/s1600/varcat_log.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="140" src="http://1.bp.blogspot.com/-Oi7ISfakhCU/VaOshgn7WsI/AAAAAAAAARc/uc62xuTPQHw/s200/varcat_log.png" width="200" /></a></div>
Figure a and b (please click to enlarge): Figure a without transformation and figure b with log(10) transformation.<br />
<br />
Code:<br />
Figure a:&nbsp; var.per.chr %&gt;% ggvis (~CHROM,~Variants, fill=~VC) %&gt;% layer_bars()<br />
Figure b: var.per.chr %&gt;% ggvis (~CHROM,~log(Variants), fill=~VC) %&gt;% layer_bars()<br />
<br />
In this figure, each bar is filled with colors as per variant category. <br />
<br />
12) List top 5 genes with highest number of variants<br />
<br />
Code: clinvar %&gt;% filter (GENEINFO!=" ") %&gt;%&nbsp; group_by (GENEINFO)&nbsp; %&gt;% summarise (Variants=n()) %&gt;% arrange(desc(Variants)) %&gt;% slice(1:5)<br />
<br />
logic: load data.frame clinvar &lt;then&gt; filter GENEINFO (gene symbol) column with no entries &lt;then&gt; group variants by gene symbol &lt;then&gt; count all the variants per gene and put in a new column titled "Variants" &lt;then&gt; arrange the genes by the number of variants per genes in descending order &lt;then&gt; display first 5 genes<br />
<br />
13) List top 5 genes with highest number of SNVs. Please note that the command can be adapted to list genes with highest MNVs and DIVs.<br />
<br />
Code: clinvar %&gt;% filter (GENEINFO!="" &amp; VC=="SNV") %&gt;%&nbsp; group_by (GENEINFO)&nbsp; %&gt;% summarise(Variants=n()) %&gt;% arrange(desc(Variants)) %&gt;% slice(1:5)<br />
<br />
14) List the gene with highest number of variants<br />
<br />
Code: clinvar%&gt;% filter (GENEINFO !=" ") %&gt;% group_by(GENEINFO) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
<br />
15)&nbsp; List the gene with highest number of SNVs, MNVs and DIVs<br />
<br />
Command for gene with highest number of<br />
<br />
SNVs:&nbsp; clinvar%&gt;% filter (GENEINFO !=" " &amp; VC=="SNV") %&gt;% group_by(GENEINFO) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
MNVs: clinvar%&gt;% filter (GENEINFO !=" " &amp; VC=="MNV") %&gt;% group_by(GENEINFO) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
DIVs: clinvar%&gt;% filter (GENEINFO !=" " &amp; VC=="DIV") %&gt;% group_by(GENEINFO) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
<br />
16) Display gene with highest number of variants on each chromosome<br />
<br />
Command: clinvar&nbsp; %&gt;% group_by(CHROM,GENEINFO) %&gt;% summarise(Genes=n()) %&gt;% filter(Genes==max(Genes))<br />
<br />
Please note that there are several entries in clinvar without gene name i.e variants might fall intergenic regions or gene deserts.<br />
<br />
17) Display the chromosome with highest number of variants that fall in genes<br />
<br />
Command:&nbsp; clinvar%&gt;% filter (GENEINFO !=" ") %&gt;% group_by(CHROM) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
<br />
18) Display the chromosome with highest number of variants<br />
<br />
Command: clinvar %&gt;% group_by(CHROM) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
<br />
19) Display the top 5 chromosomes with highest number of variants<br />
<br />
Command:&nbsp; clinvar&nbsp; %&gt;% group_by(CHROM) %&gt;% summarise(Variants=n()) %&gt;% arrange(desc(Variants)) %&gt;% slice (1:5)<br />
<br />
20) Display the top 5 genes with highest number of variants within each chromosome<br />
<br />
Command: clinvar %&gt;% filter (GENEINFO!=" ") %&gt;%&nbsp; group_by(CHROM,GENEINFO)&nbsp; %&gt;% summarise(variant=n()) %&gt;% arrange(desc(variant)) %&gt;% slice(1:5)<br />
<br />
21) Display the top gene with highest number of variants within each chromosome<br />
<br />
Command: clinvar %&gt;% filter (GENEINFO!="") %&gt;%&nbsp; group_by(CHROM,GENEINFO)&nbsp; %&gt;% summarise(variant=n()) %&gt;% filter(variant==max(variant))<br />
<br />
22) Count and display variants for each gene<br />
<br />
Command:&nbsp; clin.genes=clinvar%&gt;% filter (GENEINFO !="") %&gt;% group_by(GENEINFO) %&gt;% summarise(count=n())<br />
<br />
23) Display the first few lines<br />
<br />
Command: head(clin.genes)<br />
<br />
24) Display the number of genes on each chrosome in clinvar.vcf<br />
<br />
Command: clinvar&nbsp; %&gt;% select (CHROM, GENEINFO) %&gt;% filter (GENEINFO != "") %&gt;% distinct (GENEINFO) %&gt;% group_by(CHROM) %&gt;% summarise(Genes=n())<br />
<br />
25) Display total number of genes in clinvar. vcf<br />
<br />
Command: clinvar&nbsp; %&gt;% select (CHROM, GENEINFO) %&gt;% filter (GENEINFO != "") %&gt;% distinct (GENEINFO) %&gt;% summarise(Genes=n())<br />
<br />
26)&nbsp; Display top 5 diseases on each chromosome i.e diseases with highest number of variants in both genic and intergenic regions<br />
<br />
Command: clinvar %&gt;% filter (CLNDBN!=" ") %&gt;%&nbsp; group_by(CHROM,CLNDBN)&nbsp; %&gt;% summarise(Variants=n()) %&gt;% arrange(desc(Variants)) %&gt;% slice(1:5)<br />
<br />
27) List the disease category with highest number of variants<br />
<br />
Command:&nbsp;&nbsp; clinvar %&gt;% filter (CLNDBN!=" ") %&gt;%&nbsp; group_by(CLNDBN)&nbsp; %&gt;% summarise(Diseases=n()) %&gt;% arrange(desc(Diseases)) %&gt;% slice(1:5)<br />
<br />
<br />
Total code:<br />
<br />
# load tab file<br />
clinvar=read.delim2("clinvar_GRCh38_12jul2015.vcf.tab", header=TRUE, sep="\t")<br />
# dimensions of the file<br />
dim(clinvar)<br />
# headers of the file<br />
colnames(clinvar)<br />
# load libraries dplyr, ggvis, ggplot2<br />
lib.pac= c("dplyr", "ggplot2", "ggvis")<br />
lapply (lib.pac, require, character.only=TRUE)<br />
# total variants per chromosome<br />
total.var.per.chr=clinvar&nbsp; %&gt;% group_by(CHROM) %&gt;% summarise(n=n())<br />
# total variant categories<br />
var.cat=clinvar&nbsp; %&gt;% group_by(VC) %&gt;% summarise(Variants=n())<br />
var.cat<br />
# variant categories per chromosome<br />
var.per.chr=clinvar&nbsp; %&gt;% group_by(CHROM,VC) %&gt;% summarise(n=n())<br />
# Print only SNVs<br />
clinvar.snv=clinvar %&gt;% filter(VC=="SNV") %&gt;% group_by(CHROM,VC) %&gt;% summarise(n=n())<br />
# Visualize categories per chromosome<br />
var.per.chr %&gt;% ggvis (~CHROM,~Variants, fill=~VC) %&gt;% layer_bars()<br />
var.per.chr %&gt;% ggvis (~CHROM,~log(Variants), fill=~VC) %&gt;% layer_bars()<br />
# Top 5 genes with highest number of variants<br />
clinvar %&gt;% filter (GENEINFO!="") %&gt;%&nbsp; group_by (GENEINFO)&nbsp; %&gt;% summarise(Variants=n()) %&gt;% arrange(desc(Variants)) %&gt;% slice(1:5)<br />
# Gene with highest number of variants<br />
clinvar%&gt;% filter (GENEINFO !="") %&gt;% group_by(GENEINFO) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
# Gene with highest number of SNVs<br />
clinvar%&gt;% filter (GENEINFO !="" &amp; VC=="SNV") %&gt;% group_by(GENEINFO) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
# Gene with highest number of MNVs<br />
clinvar%&gt;% filter (GENEINFO !="" &amp; VC=="MNV") %&gt;% group_by(GENEINFO) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
# Gene with highest number of DIVs<br />
clinvar%&gt;% filter (GENEINFO !="" &amp; VC=="DIV") %&gt;% group_by(GENEINFO) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
# Display the chromsome with highest number of variants in gene regions<br />
clinvar%&gt;% filter (GENEINFO !="") %&gt;% group_by(CHROM) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
# Display the chromsome with highest number of variants <br />
clinvar %&gt;% group_by(CHROM) %&gt;% summarise(n=n()) %&gt;% filter (n==max(n))<br />
# List top 5 chromosomes with highest number of variants<br />
clinvar&nbsp; %&gt;% group_by(CHROM) %&gt;% summarise(Variants=n()) %&gt;% arrange(desc(Variants)) %&gt;% slice (1:5)<br />
# Display the top 5 genes with highest variants per each chromosome<br />
clinvar %&gt;% filter (GENEINFO!="") %&gt;%&nbsp; group_by(CHROM,GENEINFO)&nbsp; %&gt;% summarise(variant=n()) %&gt;% arrange(desc(variant)) %&gt;% slice(1:5)<br />
# Display the gene with highest variants per each chromosome<br />
clinvar %&gt;% filter (GENEINFO!="") %&gt;%&nbsp; group_by(CHROM,GENEINFO)&nbsp; %&gt;% summarise(variant=n()) %&gt;% filter(variant==max(variant))<br />
# Display number of variants per each gene<br />
clin.genes=clinvar%&gt;% filter (GENEINFO !="") %&gt;% group_by(GENEINFO) %&gt;% summarise(count=n())<br />
head(clin.genes)<br />
# Display the number of genes per chromosome<br />
clinvar&nbsp; %&gt;% select (CHROM, GENEINFO) %&gt;% filter (GENEINFO != "") %&gt;% distinct (GENEINFO) %&gt;% group_by(CHROM) %&gt;% summarise(Genes=n())<br />
# total number of genes in clinvar<br />
clinvar&nbsp; %&gt;% select (CHROM, GENEINFO) %&gt;% filter (GENEINFO != "") %&gt;% distinct (GENEINFO) %&gt;% summarise(Genes=n())<br />
#total diseases per chromosome, sorted and top 5<br />
clinvar %&gt;% filter (CLNDBN!="") %&gt;%&nbsp; group_by(CHROM,CLNDBN)&nbsp; %&gt;% summarise(Diseases=n()) %&gt;% arrange(desc(Diseases)) %&gt;% slice(1:5)<br />
#Top 5 diseases <br />
clinvar %&gt;% filter (CLNDBN!="") %&gt;%&nbsp; group_by(CLNDBN)&nbsp; %&gt;% summarise(Diseases=n()) %&gt;% arrange(desc(Diseases)) %&gt;% slice(1:5)</div>
