---
layout: post
title: Sam to fastq conversion and tag extraction
date: '2018-09-02T12:17:00.001+05:30'
author: suresh
tags:
- tags
- sam to fastq
- grep
- awk
- samtags
- sed
modified_time: '2018-09-02T12:26:45.448+05:30'
blogger_id: tag:blogger.com,1999:blog-3175511529662794199.post-2098724419655587170
blogger_orig_url: https://digibio.blogspot.com/2018/09/extract-sequence-and-tags-from-sam-file.html
---

<div style="text-align: justify;">
Researchers working with NGS data store aligned reads in a format called sam and it's binary format is bam. Both are interconvertible. Sam file has several information and for reference, please refer to sam format <a href="https://samtools.github.io/hts-specs/SAMtags.pdf">here</a> (in pdf format). One of the features of sam format is to tag sequences with appropriate tags. Some times, researchers may have to extract these tags and store it. One of the questions I came across was how to extract sequence, two tags (BZ and BQ- a phred quality score and base alignment score consecutively) and store it as fastq format. In directly, this is is equivalent to converting a sam record to fastq record.<br />
<br />
Let us do this in shell. One of the reasons I prefer scripts compared to programs is that scripts can be easily understood and can be easily fixed. But downside is that they are slow and most of the times not cross and/or backward compatible:</div>
<div style="text-align: justify;">
</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
Example of sam record:</div>
<div style="text-align: justify;">
m54071222/4194368/0_197&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 4&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; *&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 255&nbsp;&nbsp;&nbsp;&nbsp; *&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; *&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; AAGAGGAAGGGGGAGAGAGAGGAGGAGAGGGGGGAAGAGGTTGGGATGGAAA</div>
<div style="text-align: justify;">
ATAGGTGGTTAGAGGGAGAAAGG&nbsp;&nbsp; !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; np:i:1&nbsp;&nbsp; qe:i:197&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; qs:i:0&nbsp; rq:f:0&nbsp; BS:Z:ATGGCCAATTGCAGAA&nbsp;&nbsp;&nbsp; BQ:Z:JJJKKKKKKKKLLLLL&nbsp; zm:i:4194368&nbsp;&nbsp;&nbsp; RG:Z:1f1bf15c&nbsp;&nbsp; sc:A:L&nbsp; sz:A:N</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
You can download the example sam with one record <a href="https://drive.google.com/file/d/19xohmMFLIMuctmMVY3SUAXyCZJxmujU3/view?usp=sharing">here</a>. Note that tags can appear in random order. For convenience, command line script is broken into multilines:</div>
<div style="text-align: justify;">
============================================================ </div>
<div style="text-align: justify;">
$ paste&nbsp; &lt;(awk '{print $1}' test.sam) \&nbsp;</div>
<div style="text-align: justify;">
&nbsp;&lt;( grep -wo "\&lt;BS\W\w\W\w\+\&gt;" test.sam) \</div>
<div style="text-align: justify;">
&nbsp;&lt;(grep -wo "\&lt;BQ\W\w\W\w\+\&gt;" test.sam) | sed 's/:/\t/g' | \</div>
<div style="text-align: justify;">
&nbsp;awk -v OFS="\n" -v ORS="\n\n" '{print "@"$1,$4,"+",$7}'</div>
<div style="text-align: justify;">
output: </div>
<div style="text-align: justify;">
@m54071222/4194368/0_197<br />
ATGGCCAATTGCAGAA<br />
+<br />
JJJKKKKKKKKLLLLL<br />
<br />
@m54071222/4194368/0_197<br />
ATGGCCAATTGCAGAA<br />
+<br />
JJJKKKKKKKKLLLLL<br />
================================================================</div>
<div style="text-align: justify;">
This function/ command has several things going on. Let me explain it here:</div>
<ol>
<li>Script has 3 actions: paste, replacement by sed and output formatting by awk</li>
<li>paste command takes two grep functions and one awk function.</li>
<li>Grep functions are grepping for tags (BS and BQ) using word (-w) argument, one time for each argument</li>
<li>Awk function is printing first column (read ID)</li>
<li>Sed function is breaking down the tags by full colon so that tags are broken into 3 pieces (from BS:Z:ATGGCCAATTGCAGAA to BS, Z and ATGGCCAATTGCAGAA)</li>
<li>Use awk to extract necessary columns. Append @ to the ID, print BS sequence, print + and finally BZ sequence. Each field (column) will be separated by newline so that each line (from sam) is converted to fastq and each record (each line in sam) will be separated by two new lines. User can remove this two lines if necessary. Logic was to insert to two lines separating each fastq record, for better reading.</li>
</ol>
<div style="text-align: justify;">
I think this is the logic behind sam2fastq function if you ever want to convert your sam file to fastq format using any of the fields. However remember that this is for single ended, not for paired sequencing. Note that when you convert a sam record to fastq format, it must adhere fastq format standards so that downstream tools can deal with fastq files appropriately. </div>
<div style="text-align: justify;">
<br /></div>
