<div dir="ltr" style="text-align: left;" trbidi="on">
<div style="text-align: justify;">
Transcriptomics has evolved over the years from isolated, targeted genes to <i>enmasse&nbsp; </i>studies of transcripts. This allowed scientists not only to study the transcriptome it self, but it's dynamics. One of the <i>enmasse</i> technologies that led the revolution is array based technology that realized the dreams of scientist to study entire transcriptome in it's context. Currently, NGS based technologies replaced use of arrays. However, this is limited labs with enough grants and probably established labs. There are several labs still using microarrays to study transcriptome. Transcritomic microarrays are largely classified into 2 categories: 3' IVT arrays and Exon arrays. 3'IVT arrays dominated early where as exon arrays slowly took over 3' IVT arrays. Information from exon arrays is multifold compared to 3' IVT microarrays. It is interesting to note that Affymetrix dominated early market (esp 3' IVT market), late entrant Illumina took over the market esp from exon arrays onwards and consolidated its position with SNP (genomic) chips and NGS technologies.&nbsp;</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
In this note, analysis of GSE-45161 (HGU133plus2 human chip) data analysis would be presented. User can download the data from <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45161" target="_blank">here</a> or&nbsp; user can download the data in to R directly, following this <a href="http://digibio.blogspot.com/2015/09/download-and-use-microarray-files-from.html" target="_blank">note</a>.&nbsp;&nbsp; HGU133plus2 is produced and marketed by Affymetrix. We will use limma in R/Bioconductor for&nbsp; analysis.&nbsp; Please note following information:<br />
<ul>
<li>Example files used in the workflow are arbitrarily taken to demo the workflow</li>
<li>Analysis results from the example workflow are mostly unscientific and cannot be used in scientific deduction from this analysis</li>
</ul>
There are several tutorials scattered all over the internet on HGU133plus2 data analysis.&nbsp; Here I am trying to collate transcriptomics data analysis methods in this blog. Hence this attempt is made to revisit the microarray data analysis.<br />
<br />
Following is the typical analysis of microarray data:<br />
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
<a href="http://2.bp.blogspot.com/--mqGORkZ2Vc/VfJVp7WyZYI/AAAAAAAAAY0/OTTDPC_GpSw/s1600/hgu133plus2%25281%2529.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="156" src="http://2.bp.blogspot.com/--mqGORkZ2Vc/VfJVp7WyZYI/AAAAAAAAAY0/OTTDPC_GpSw/s640/hgu133plus2%25281%2529.png" width="640" /></a></div>
<div class="separator" style="clear: both; text-align: center;">
<a href="http://4.bp.blogspot.com/-6qxAmlI9crw/VfJRgLLcNMI/AAAAAAAAAYo/tih-KfHv4a8/s1600/hgu133plus2.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><br /></a></div>
&nbsp;Analysis can be largely broken down into 4 stages as explained below:<br />
<ul>
<li>Raw files</li>
</ul>
Microarray experiments in general carried out by either core facilities or commercial providers. In either case, raw files are provided as CEL and CHP&nbsp; files. Since scientist knows better what to look for, it is better to start working with cel files instead of relying on excel/analyzed files provided by service providers. <br />
<ul>
<li>Quality check</li>
</ul>
There are several tools provided by bioconductor to check&nbsp; the quality of the data. Collated quality check package is <a href="http://www.bioconductor.org/packages/release/bioc/html/affyQCReport.html" target="_blank">AffyQCreport</a>. In general, there are several quality checks that assist user in taking decisions. Common quality checks are box plots (for raw data distribution), density plots (intensity distribution), RNA degradation plots, MA plots, raw image plots, pseudo color plots etc. In this notes, boxplots before and after normalization, RNA degradation plots, MA plots, raw image plots, internal controls (GAPDH and beta-actin) are presented.</div>
<div style="text-align: justify;">
<ul>
<li>Data analysis</li>
</ul>
For the brevity of this note, normalization, experimental design, contrast calculation, model fit and calculation of deferentially expressed probesets are grouped under statistics, though last two steps are related to statistics (model and calculation of differential expression). Differential analysis is basically filtering of data based on several criteria (p-value, F statistic, fold change, SD across samples etc). This also involves visualization of analyzed data using heatmaps, treemaps, clustering of normalized data, MA plots, boxplots etc.&nbsp;&nbsp;</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
Please note that three normalized methods (RMA, GCRMA and VSN) in this note. However, subsequent analysis use GCRMA treated data. User can use any one of the above 3 methods. This was done as I observed discrepancies in their clustering, post normalization.</div>
<div style="text-align: justify;">
<ul>
<li>Functional analysis&nbsp;</li>
</ul>
</div>
<div style="text-align: justify;">
Functional analysis involves identification of significant genes, GO analysis, pathway analysis, network analysis, Interaction networks etc. In this note, we present, GO enrichment as example.&nbsp;</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
After downloading the data either using browser or R as mentioned above, run the following script. Please keep the files in a folder titled "gse45161" in the same directory or change the code as per your raw files location and download the sample description file <a href="https://docs.google.com/spreadsheets/d/1wfB4LeBQbx9hzL4RX9n6NzviqsJtFv-Nh3ji5EQVah0/edit?usp=sharing" target="_blank">here</a> and save it as tsv with name "GSE45161.csv". If you are using Rstudio, image command may throw an error saying that figure margins are too large. Try to clear existing plots or expand the plots window. :</div>
<div style="text-align: justify;">
<br /></div>
<div style="text-align: justify;">
###########################################</div>
<div style="text-align: justify;">
#######R script starts here######################</div>
<div style="text-align: justify;">
############################################</div>
library(affy)<br />library(gcrma)<br />library(limma)<br />library(treemap)<br />library(gplots)<br />library(simpleaffy)<br />library(RColorBrewer)<br />library(hgu133plus2.db)<br />library(treemap)<br />library(vsn)<br />library(GO.db)<br />#load("gse45161.Rdata")<br />ls()<br />getwd()<br />#setwd("/wine/gse45161/")<br />###read experimental data<br />samples.cel=data.frame(read.delim2("GSE45161.csv", row.names = 2, sep="\t", header=TRUE))<br />### sort row names (sample names)<br />samples.cel=samples.cel[sort(row.names(samples.cel)),]<br />#samples.cel<br />### load cel files and store as object<br />GSE45161.cel=ReadAffy(filenames = row.names(samples.cel))<br />#colnames(GSE45161.cel)<br /># ## Regenerate raw images<br />png("GSE45161.raw.png")<br />par(mfrow=c(2,5))<br />image(GSE45161.cel)<br />dev.off()<br /># ## Quality control####<br />qc.GSE45161.cel=qc(GSE45161.cel)<br /># ## Plot quality controls<br />png("GSE45161.cel.qc.png",width=1366, height=768)<br />plot(qc.GSE45161.cel)<br />dev.off()<br />### Plot quality controls with mid probes<br />png("GSE45161.cel.qc.mid.png", width=1366, height=768)<br />plot(qc.GSE45161.cel, usemid=T, type=c)<br />dev.off()<br />## RNA degradation<br />rnadeg.GSE45161=AffyRNAdeg(GSE45161.cel)<br />cols=1:10<br />png("GSE45161.rnadeg.png", width=1366, height=768)<br />plotAffyRNAdeg(rnadeg.GSE45161,cols =cols )<br />legend("topleft", box.col = c(1,2), legend=samples.cel$Name,col=cols, lty=1, y.intersp =1, cex=2, xjust = 0)<br />dev.off()<br />## Histograms for raw data<br />png("GSE45161.hist.png", width=1366, height=768, res=90)<br />hist(GSE45161.cel)<br />legend("topright", box.col = c(1,2), legend=samples.cel$Name,col=cols, lty=1, y.intersp =1, cex=2, xjust = 0)<br />dev.off()<br />### Normalize (GCRMA) cel files<br />gc.GSE45161=gcrma(GSE45161.cel)<br />rma.GSE45161=rma(GSE45161.cel)<br />vsn.GSE45161=vsnrma(GSE45161.cel)<br /># Plot meanSD plot for VSN normalized values<br />#class(vsn.GSE45161)<br />png("vsn.meansd.png")<br />meanSdPlot(vsn.GSE45161)<br />dev.off()<br />### check if samples names are same between normalized values and samples<br />all(colnames(gc.GSE45161)==rownames(samples.cel))<br />### Store the condition<br />condition=samples.cel$Target<br />class(condition)<br />### design model matrix<br />design=model.matrix(~0+condition)<br />colnames(design)<br />colnames(design)=c("Avastin","control")<br />design<br />### Fit to a model - linear model<br />fit=lmFit(gc.GSE45161,design)<br />names(fit)<br />fit$design<br />### make contrasts<br />cm=makeContrasts(Avastin-control,levels=design)<br />#### contrast fit<br />cmfit=contrasts.fit(fit, cm)<br /># ebayes fit<br />ecmfit=eBayes(cmfit)<br />names(ecmfit)<br />#MA plot<br />png("gse45161_maplot.png", width=1366, height=768)<br />plot(ecmfit$Amean, ecmfit$coefficients[,1], cex=0.3, col="blue")<br />dev.off()<br />### volcano plot<br />png("gse45161_volcanoplot.png", width=1366, height=768)<br />plot (ecmfit$coefficients[,1], -log10(ecmfit$p.value[,1]), pch=20, cex=0.3)<br />dev.off()<br />### boxplot<br />png("gse45161_boxplot_before_after.png", width=1366, height=768)<br />par(mfrow=c(1,2))<br />boxplot(GSE45161.cel, main="Before normalization", las=3, names=samples.cel$Name, notch=TRUE, cex.axis=0.7)<br />boxplot(data.frame(exprs(gc.GSE45161)), main="After normalization", las=3, names=samples.cel$Name, notch=TRUE, cex.axis=0.7)<br />dev.off()<br />## write to HDD<br />write.fit(ecmfit, file="fitted_expression.txt", method = "separate")<br />write.table(as.data.frame(ecmfit), file="gse45161_expression.txt", sep="\t")<br />#####Extract genes/probes of interest<br /># filter probes intensity &gt;6 and fold change &gt; 2 or &lt;-2<br />afc.ecmfilt=ecmfit[subset(ecmfit$Amean&gt; 6 &amp; abs(ecmfit$coefficients)&gt; 2),]<br />#class(afc.ecmfilt)<br />#names(afc.ecmfilt)<br />#pecmfilt$Amean<br />#MA plot of total and highlight genes with fold change of 2 and A &gt;6<br /># MA plot for ebayes fit probes<br />plot (ecmfit$Amean, ecmfit$coefficients, cex=0.3, xlab="Intensity change (log)", ylab="Fold change (log)")<br /># high light genes of intereste<br />points(afc.ecmfilt$Amean, afc.ecmfilt$coefficients, col="red")<br />dev.off()<br />### volcano plot<br /># Volcano plot for ebayes fit probes (foldchange Vs P value)<br />png("volcano.png")<br />plot (ecmfit$coefficients, -log10(ecmfit$p.value), pch=20, cex=0.3, xlim=c(3,-3), xlab= "Fold change (log)", ylab="P value (-log10P)")<br />points (afc.ecmfilt$coefficients, -log10(afc.ecmfilt$p.value), pch=20,xlim=c(3,-3), col="green")<br />dev.off()<br />#############################################<br />#####clutering###############################<br />#############################################<br />### clustering of gcrma normalized data<br />dat=exprs(gc.GSE45161)<br />dist.dat=dist(t(dat))<br />hc.dist.dat=hclust(dist.dat, method="complete")<br />png("gc.clustering.png")<br />plot(hc.dist.dat, labels=paste0(samples.cel$Name, sep=",", samples.cel$Target))<br />dev.off()<br />### clustering of RMA normalized data<br />rma.dat=exprs(rma.GSE45161)<br />rma.dist.dat=dist(t(rma.dat))<br />#image(as.matrix(rma.dist.dat))<br />rma.hc.dist.dat=hclust(rma.dist.dat, method="complete")<br />png("rma.clustering.png")<br />plot(rma.hc.dist.dat, labels=paste0(samples.cel$Name, sep=",", samples.cel$Target))<br />dev.off()<br />### clustering of vsn normalized data<br />vsn.dat=exprs(vsn.GSE45161)<br />vsn.dist.dat=dist(t(vsn.dat))<br />#image(as.matrix(vsn.dist.dat))<br />vsn.hc.dist.dat=hclust(vsn.dist.dat, method="complete")<br />png("vsn.clustering.png")<br />plot(vsn.hc.dist.dat, labels=paste0(samples.cel$Name, sep=",", samples.cel$Target))<br />dev.off()<br />###############################################<br /># Use hgu133plus2 db for extracting information<br />##############################################<br />probes=row.names(ecmfit)<br />head(probes)<br />### using hgu133plus2.db<br />columns(hgu133plus2.db)<br /># Extract probe positions, gene symbol, pathway information<br />mapping=select(hgu133plus2.db, keys=probes,columns=c("PROBEID", "CHRLOC", "CHRLOCEND","SYMBOL"), keytype ="PROBEID")<br />head(mapping)<br />dim(mapping)<br />pruned.mapping=na.omit(mapping[grep('un|ssto|hap|random',mapping$CHRLOCCHR, invert = TRUE, ignore.case = TRUE),])<br />#######################<br /># Further processing-- Annotation<br />########################<br />#Merge probes and data set<br />mecmfit=merge(ecmfit,pruned.mapping, by.x=0, by.y="PROBEID")<br />head(mecmfit)<br />dim(mecmfit)<br />## Change column names for better understanding<br />colnames(mecmfit)[c(1,2,13:17)]=c("Probe","FoldChange", "F.p.value", "START", "CHR", "END", "SYMBOL" )<br />### reorder columns<br />mecmfit=mecmfit[,c(1,2,17,15,14,16,3:13)]<br />head(mecmfit)<br />### sort by chromosome and position<br />ormecmfit=mecmfit[order(mecmfit$CHR, abs(mecmfit$START), decreasing = FALSE),]<br />ormecmfit$CHR=as.factor(ormecmfit$CHR)<br />dim(ormecmfit)<br />write.table(ormecmfit, "foldchange.txt", sep="\t", row.names =F)<br />dim(unique(ormecmfit))<br />### filter probeset with more than 2 fold change<br />fc.ormecmfit=ormecmfit[abs(ormecmfit$FoldChange)&gt;2,]<br />#View(fc.ormecmfit)<br /># Extract probe sets with 1.5 FC across all the chips<br />f_ormecmfit=subset(ormecmfit, abs(ormecmfit$FoldChange)&gt;1.5)<br />dim(f_ormecmfit)<br />tm.uf_ormecmfit=f_ormecmfit[,c(1,2,3,14)]<br />dim(tm.uf_ormecmfit)<br /># Unique probesets<br />utm.uf_ormecmfit=unique(tm.uf_ormecmfit, by=tm.uf_ormecmfit$Probe)<br />#dim(utm.uf_ormecmfit)<br />#class(utm.uf_ormecmfit)<br />#View(utm.uf_ormecmfit)<br />### treemap <br />png("treemap.png")<br />treemap(utm.uf_ormecmfit, index=c("SYMBOL"), type="index", vSize ="p.value", vColor ="FoldChange" , palette ="RdGy")<br />dev.off()<br />??aafTableAnn<br />##GO<br />go.utm.uf_ormecmfit=select(hgu133plus2.db, utm.uf_ormecmfit$Probe, "GO", "PROBEID")<br />go.term.utm.uf_ormecmfit= select(GO.db, go.utm.uf_ormecmfit$GO,"TERM","GOID")<br />####&nbsp; Probes per chromosome and store in a data frame<br /># table(ormecmfit$CHR)<br />df.ormecmfit=as.data.frame(table(ormecmfit$CHR))<br />## Plot frequency of probes per chromosome<br />png("hist_perchromosome.png")<br />barplot(df.ormecmfit$Freq,&nbsp; xlab="Chromosomes", names.arg=df.ormecmfit$Var1, ylab="Probset number")<br />dev.off()<br />#save.image("gse45161.Rdata")<br /><br />################################################################<br />
<div style="text-align: justify;">
####script ends######################</div>
<div style="text-align: justify;">
########################################################</div>
<div style="text-align: justify;">
<br />
Few images derived out of analysis are collated below:<br />
<br />
<div class="separator" style="clear: both; text-align: center;">
<a href="http://3.bp.blogspot.com/-JIzI7kFtSZc/VfJev-ejfTI/AAAAAAAAAZE/keZUYDwyKE0/s1600/Capture.PNG" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="305" src="http://3.bp.blogspot.com/-JIzI7kFtSZc/VfJev-ejfTI/AAAAAAAAAZE/keZUYDwyKE0/s640/Capture.PNG" width="640" /></a></div>
<br />
<br />
<br /></div>
</div>
